src/components/Menu/ControlledMenu.js

```diff
@@ -104,24 +99,13 @@ class ControlledMenu extends Component {
   };
 
   renderChildren = () => {
-    const { children, checkbox } = this.props;
+    const { children } = this.props;
     return React.Children.map(children, (child, index) => {
       const id = getUniqueHash('item', index);
 
       if (!child) return null;
 
-      if (child.type !== MenuItem) {
-        return React.cloneElement(child, {
-          key: id
-        });
-      }
-
-      return React.cloneElement(
-        child, {
-          selected: checkbox ? child.props.selected === true : null,
-          key: id,
-        }
-      );
+      return React.cloneElement(child, { key: id });
```

Comment:
IIRC there is a reason for this check - I will look into it tomorrow. Not sure what this accomplishes and why it's failing.

Reply:
Update: This check was apparently added to distinguish between different child accessibility roles. It abuses the boolean `selected` by assigning a semantic meaning to the value `null`. The assignment `checkbox ? child.props.selected === true : null` accomplishes an explicit cast to a bool. value when the `checkbox` prop is truthy, otherwise it explicitely casts to `null`.

@PeterSiegl This behavior is correct from an accessibility standpoint as far as I can tell from looking at the LDS specification. The implementation seems hacky. Does this cause an unrelated bug for you?

Reply:
@fspoettel it's not really a bug the different behavior between production and localhost is just confusing. for production I have to set the `checkbox` prop which is not really obvious since locally everything works fine.